cohorts/variant_stats.py

```diff
@@ -158,6 +160,30 @@ def maf_somatic_variant_stats(variant, variant_metadata):
         normal_stats = _maf_variant_stats(variant, variant_metadata, prefix="n")
     return SomaticVariantStats(tumor_stats=tumor_stats, normal_stats=normal_stats)
 
+def _vcf_is_strelka(variant_file):
+    """Return True if variant_file given is in strelka format
+    """
+    return "strelka" in variant_file.lower()
+
+def _vcf_is_maf(variant_file):
+    """Retrun True if variant_file given is in .maf format
+    """
+    return ".maf" in variant_file.lower()
+
+def _vcf_is_mutect(variant_file):
+    """Return True if variant_file give is in mutect format
+    """
+    if "mutect" in variant_file.lower():
+        return True
+    else:
+        try:
+            vcf_reader = vcf.Reader(open(variant_file, "r"))
```

Comment:
These are different. the variant_metadata is not the same thing as the file_metadata, afaict.

E.g.:
```python
(Pdb) variant_metadata.keys()
dict_keys(['id', 'filter', 'qual', 'info', 'sample_info', 'alt_allele_index'])
(Pdb) vcf_reader = vcf.Reader(open(variant_file, "r"))
(Pdb) vcf_reader.metadata.keys()
odict_keys(['fileformat', 'GATKCommandLine.CombineVariants', 'cmdline', 'content', 'fileDate', 'germlineIndelTheta', 'germlineSnvTheta', 'priorSomaticIndelRate', 'priorSomaticSnvRate', 'reference', 'source', 'source_version', 'startTime', 'SnpEffVersion', 'SnpEffCmd'])
```
Oddly, this dict contains very different things even for files generated from the same epidisco pipeline. 

for example, in this case the vcf has a metadata field which is particularly helpful:

```python
(Pdb) vcf_reader.metadata['content']
['strelka somatic indel calls']
```

I didn't look for particular fields based mostly on arman's feedback, which suggested that the set of fields may be different for different versions within a particular format. Also trying to read the file according to mutect or strelka format would implicitly test for presence of fields. So this is the fall-back mechanism in a way.